assume plus example interacts badly with pytest.mark.parametrize

I can see why I see hypothesis.errors.UnsatisfiedAssumption when I supply an example that fails to satisfy the assumption, though it is inconvenient (I prefer to leave old failures lying around in @example in case I want to remove the assumption later). But it seems like @example is not respecting the parametrized argument it contains? That is, the hypothesis-generated parts are being combined with both parametrize values, and when the example doesn't pass the assumption, I see UnsatisfiedAssumption - even though it is for a combination of values I did not specify.  

Sorry, that's not a very clear explanation. But perhaps this contrived example will help:

```python
import pytest
from hypothesis import assume, given, example
from hypothesis.strategies import floats


@given(floats(0, 1))
@example(t=1, f=0.56789)
@pytest.mark.parametrize("t", [0.5, 1])
def test_unsatisfied_assumption(t, f):
    assume(f < t)
    assert f < t
```
yields
```
E           hypothesis.errors.UnsatisfiedAssumption
```
and
```
Falsifying example: test_unsatisfied_assumption(t=0.5, f=0.56789)
```

--- a/hypothesis-python/RELEASE.rst
+++ b/hypothesis-python/RELEASE.rst
@@ -1,0 +1,14 @@
+RELEASE_TYPE: minor
+
+This release improves the interaction between :func:`~hypothesis.assume`
+and the :func:`@example() <hypothesis.example>` decorator, so that the
+following test no longer fails with ``UnsatisfiedAssumption`` (:issue:`2125`):
+
+.. code-block:: python
+
+    @given(value=floats(0, 1))
+    @example(value=0.56789)  # used to make the test fail!
+    @pytest.mark.parametrize("threshold", [0.5, 1])
+    def test_foo(threshold, value):
+        assume(value < threshold)
+        ...

--- a/hypothesis-python/src/hypothesis/core.py
+++ b/hypothesis-python/src/hypothesis/core.py
@@ -347,6 +347,13 @@ def execute_explicit_examples(state, wrapped_test, arguments, kwargs):
                         is_final=True,
                         print_example=True,
                     )
+            except UnsatisfiedAssumption:
+                # Odd though it seems, we deliberately support explicit examples that
+                # are then rejected by a call to `assume()`.  As well as iterative
+                # development, this is rather useful to replay Hypothesis' part of
+                # a saved failure when other arguments are supplied by e.g. pytest.
+                # See https://github.com/HypothesisWorks/hypothesis/issues/2125
+                pass
             except BaseException:
                 report_buffered()
                 raise

--- a/hypothesis-python/tests/cover/test_explicit_examples.py
+++ b/hypothesis-python/tests/cover/test_explicit_examples.py
@@ -18,9 +18,18 @@ from unittest import TestCase
 
 import pytest
 
-from hypothesis import Phase, Verbosity, example, given, note, reporting, settings
+from hypothesis import (
+    Phase,
+    Verbosity,
+    assume,
+    example,
+    given,
+    note,
+    reporting,
+    settings,
+)
 from hypothesis.errors import DeadlineExceeded, InvalidArgument
-from hypothesis.strategies import integers, nothing, text
+from hypothesis.strategies import floats, integers, nothing, text
 from tests.common.utils import assert_falsifying_output, capture_out
 
 
@@ -228,3 +237,12 @@ def test_runs_deadline_for_examples():
 
     with pytest.raises(DeadlineExceeded):
         test()
+
+
+@given(value=floats(0, 1))
+@example(value=0.56789)
+@pytest.mark.parametrize("threshold", [0.5, 1])
+def test_unsatisfied_assumption_during_explicit_example(threshold, value):
+    # Regression test, expected to pass / skip depending on parametrize.
+    # See https://github.com/HypothesisWorks/hypothesis/issues/2125
+    assume(value < threshold)

